File: elixir/test/merchant_portal_common_controller_entity_test.exs
# MerchantPortalCommonController entity test (offline, mock transport)

defmodule BluefinTecsMerchantPortal.MerchantPortalCommonControllerEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias BluefinTecsMerchantPortal.Helpers, as: H
  alias BluefinTecsMerchantPortal.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/merchant_portal_common_controller/MerchantPortalCommonControllerTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    BluefinTecsMerchantPortal.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.merchant_portal_common_controller"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = BluefinTecsMerchantPortal.test()
    ent = BluefinTecsMerchantPortal.merchant_portal_common_controller(sdk)
    assert ent != nil
  end

  test "should load an existing record" do
    id = first_id()

    if id != nil do
      sdk = mk_sdk()
      ent = BluefinTecsMerchantPortal.merchant_portal_common_controller(sdk)
      rec = BluefinTecsMerchantPortal.Entity.MerchantPortalCommonController.load(ent, S.jm(["id", id]))
      assert S.ismap(rec)
      assert S.getprop(rec, "id") == id
    end
  end
end

File: scala/sdktest/MerchantPortalCommonControllerEntityTest.scala
// Generated basic-flow test for the merchant_portal_common_controller entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped MerchantPortalCommonControllerTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsmerchantportalsdk.core.{Helpers, BluefinTecsMerchantPortalSDK}
import voxgig.bluefintecsmerchantportalsdk.utility.struct.Struct

object MerchantPortalCommonControllerEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("merchant_portal_common_controller.instance") {
      val testsdk = BluefinTecsMerchantPortalSDK.testSDK()
      val ent = testsdk.merchantPortalCommonController(null)
      rep.check("merchant_portal_common_controller.instance", ent != null, "expected non-null merchant_portal_common_controller entity")
    }

    rep.scope("merchant_portal_common_controller.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/merchant_portal_common_controller/MerchantPortalCommonControllerTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsMerchantPortalSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("merchant_portal_common_controller01", "MERCHANT_PORTAL_COMMON_CONTROLLER01")
      idmap.put("merchant_portal_common_controller02", "MERCHANT_PORTAL_COMMON_CONTROLLER02")
      idmap.put("merchant_portal_common_controller03", "MERCHANT_PORTAL_COMMON_CONTROLLER03")
      val now = System.currentTimeMillis()

      // LOAD
      val merchantPortalCommonControllerRef01Ent = client.merchantPortalCommonController(null)
      val merchantPortalCommonControllerRef01MatchDt0 = new LinkedHashMap[String, Object]()
      val merchantPortalCommonControllerRef01DataDt0Loaded = merchantPortalCommonControllerRef01Ent.load(merchantPortalCommonControllerRef01MatchDt0, null)
      rep.check("merchant_portal_common_controller.load.nonnull", merchantPortalCommonControllerRef01DataDt0Loaded != null, "expected load result to be non-null")
    }
  }
}
